inspect_byproducts.py

```diff
@@ -172,5 +184,6 @@ def main():
       inspect_byproducts(args.link, args.outerr, args.operator, args.string)
 
 
+
```

Comment:
Returning an `int` from `main()` won't be visible on the command line. Use `sys.exit(<return code>)` instead, or if you want to be really cool, you could use `ArgumentParser.error(<message>)` in case a wrong argument was passed, which also prints the usage message.

As for `inspect_byproducts`, i would do something like:
```python
try:
  if inspect_byproducts(args.link, args.outerr, args.operator, args.string):
    # Everything went fine
    sys.exit(0) 
  else:
    # The inspection failed
    sys.exit(1) 

# We could even be more granular and exit with different codes for `KeyError`, `IOError` and other Exceptions
except Exception as e:
  sys.exit(<some value greater than 2 and lower than 127>)
```